lib/fizzy/execute.cpp

```diff
@@ -524,6 +524,45 @@ inline uint64_t popcnt64(uint64_t value) noexcept
     return static_cast<uint64_t>(__builtin_popcountll(value));
 }
 
+template <typename T>
+inline T fceil(T value) noexcept
+{
+    static_assert(std::is_floating_point_v<T>);
+    if (std::isnan(value))
+        return std::numeric_limits<T>::quiet_NaN();  // Positive canonical NaN.
+
+    // The FE_INEXACT error is ignored (whenever the implementation reports it at all).
+    return std::ceil(value);
+}
+
+template <typename T>
+inline T ffloor(T value) noexcept
+{
+    static_assert(std::is_floating_point_v<T>);
+    if (std::isnan(value))
+        return std::numeric_limits<T>::quiet_NaN();  // Positive canonical NaN.
+
+    // The FE_INEXACT error is ignored (whenever the implementation reports it at all).
+    const auto result = std::floor(value);
+
+    // TODO: GCC / glibc BUG WORKAROUND:
+    // When rounding direction is set to FE_DOWNWARD the std::floor outputs -0 where it should +0.
+    // The following workarounds the issue by using the fact that the sign of the output must always
+    // match the sign of the input value.
+    return std::copysign(result, value);
```

Comment:
Perhaps also leave the gcc bug link here (I know it was reported for ppc, but this seems similar).

Reply:
That one is fixed. We need to report new one, but we need a reproducer first.

Reply:
`ceil` that results in `-0` doesn't have this problem?

Reply:
`ceil` is tested the same way, and no issues.

Do you mean `ceil()` returning +0 where it should -0? The test cases of values from `[-1,-0]` are included.